pkg/fqdn/matchpattern/matchpattern.go

```diff
@@ -24,7 +25,7 @@ func Validate(pattern string) (matcher *regexp.Regexp, err error) {
 		return nil, errors.New(`Only alphanumeric ASCII characters, the hyphen "-", underscore "_", "." and "*" are allowed in a matchPattern`)
 	}
 
-	return regexp.Compile(ToRegexp(pattern))
+	return re.CompileRegex(ToRegexp(pattern))
```

Comment:
Maybe worth calling it out here, but reviewers might wonder: how do we know if this will even use the LRU, i.e. if the LRU even initialized at the time this code path executes?

The answer is yes considering that this code path is called to "sanitize" the policies (read validate) upon submission from the user.

The LRU is initialized during dameon bootstrap of the FQDN subsystem which happens before the K8s watcher sync starts (`InitK8sSubsystem()`).

Reply:
Happy to add this as a comment somewhere or maybe inside the commit msg, once CI finishes.

Reply:
Ah hehe, exactly the topic of my feedback below: https://github.com/cilium/cilium/pull/19632#discussion_r862192900